cypress/e2e/utils/sharedTests/changeBranch.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  newClickFlow,
+  newExpectation,
+  newExpectationWithClickFlows,
+  newExpectationWithURL,
+  newShouldArgs,
+} from "../helpers";
+
+const beVisible = newShouldArgs("be.visible");
+const beVisibleAndContain = (value: string) =>
+  newShouldArgs("be.visible.and.contain", value);
+
+type TestParams = {
+  isLeftNavClosed: boolean;
+  currentTabID: string;
+  destinationURL: string;
+  optionalText?: string;
+  destinationBranch: string;
+  destinationBranchOptionalText?: string;
+};
+
+const checkCurrentBranch = (testParams: TestParams) =>
+  testParams.destinationBranchOptionalText
+    ? [
+        newExpectationWithURL(
+          "should have page load in ",
+          `[data-cy=${testParams.currentTabID}]`,
+          beVisible,
+          `${testParams.destinationURL}`,
+        ),
+        newExpectation(
+          "should see new branch ",
+          `[data-cy=${testParams.currentTabID}]`,
+          beVisibleAndContain(testParams.destinationBranchOptionalText),
+        ),
+      ]
+    : [
+        newExpectationWithURL(
+          "should have page load in ",
+          `[data-cy=${testParams.currentTabID}]`,
+          beVisible,
+          `${testParams.destinationURL}`,
+        ),
+      ];
+
+export const changeBranch = (testParams: TestParams) => [
```

Comment:
Helps to have return types on all of these
```suggestion
export const changeBranch = (testParams: TestParams): Tests => [
```